library/core/class.format.php

```diff
@@ -1430,6 +1430,52 @@ public static function Mentions($Mixed) {
       }
    }
 
+   /** Do a preg_replace, but don't affect things inside <code> tags.
+    * The three parameters are identical to the ones you'd pass
+    * preg_replace.
+    *
+    * @param mixed $Pattern
+    * @param mixed $Replacement
+    * @param mixed $Subject
+    * @param bool $IsCallback If true, do preg_replace_callback. Do
+    *             preg_replace otherwise.
+    * @return string
+    */
+   public static function ReplaceButProtectCodeBlocks($Pattern, $Replacement, $Subject, $IsCallback = FALSE) {
+     // Take the code blocks out, replace with something unlikely to
+     // appear in the string, and keep track of what substring got
+     // replaced with what code.
+     $Replacements = array();
+     $Subject = preg_replace_callback(
+       '/<code>.*?<\/code>/i',
+       function($Matches) use (&$Replacements) {
+         // Random string and replacement index, surrounded by
+         // whitespace to try to prevent the characters from being
+         // picked up by $Pattern.
+         $ReplacementString = ' 5z9Ah9Y2sX5xnzUx8wSq'.count($Replacements).' ';
```

Comment:
Instead of random-string-plus-running-total-of-replacements, why not use [sha1](http://php.net/sha1)?  You can use the hash as $ReplacementString.  You also wouldn't need to reverse your placeholder array at the end of your function.